launchable/commands/subset.py

```diff
@@ -526,88 +534,112 @@ def get_payload(
+                # With non-blocking mode, we don't need to wait for the response
+                sys.exit(0)
+
+            try:
+                res = subset_request(client=client, timeout=timeout, payload=payload)
+                # The status code 422 is returned when validation error of the test mapping file occurs.
+                if res.status_code == 422:
+                    msg = "Error: {}".format(res.reason)
+                    tracking_client.send_error_event(
+                        event_name=Tracking.ErrorEvent.USER_ERROR,
+                        stack_trace=msg,
+                    )
+                    click.echo(
+                        click.style(msg, fg="red"),
+                        err=True)
+                    sys.exit(1)
+
+                return SubsetResult.from_response(res.json())
+            except Exception as e:
+                tracking_client.send_error_event(
+                    event_name=Tracking.ErrorEvent.INTERNAL_CLI_ERROR,
+                    stack_trace=str(e),
+                )
+                client.print_exception_and_recover(
+                    e, "Warning: the service failed to subset. Falling back to running all tests")
+                return SubsetResult.from_test_paths(self.test_paths)
+
         def run(self):
             """called after tests are scanned to compute the optimized order"""
-            if not is_get_tests_from_previous_sessions and len(self.test_paths) == 0:
+            if not self.is_get_tests_from_previous_sessions and len(self.test_paths) == 0:
+                msg = None
                 if self.input_given:
                     msg = "ERROR: Given arguments did not match any tests. They appear to be incorrect/non-existent."  # noqa E501
+                elif client.is_enabled_pts_v2():
+                    click.echo("INFO: Subset input is empty, enabling `--get-tests-from-previous-sessions` option", err=True)
+                    self.is_get_tests_from_previous_sessions = True
                 else:
                     msg = "ERROR: Expecting tests to be given, but none provided. See https://www.launchableinc.com/docs/features/predictive-test-selection/requesting-and-running-a-subset-of-tests/subsetting-with-the-launchable-cli/ and provide ones, or use the `--get-tests-from-previous-sessions` option"  # noqa E501
-                click.echo(click.style(msg, fg="red"), err=True)
-                exit(1)
 
-            # When Error occurs, return the test name as it is passed.
-            original_subset = self.test_paths
-            original_rests = []
-            summary = {}
-            subset_id = ""
-            is_brainless = False
-            is_observation = False
+                if msg:
+                    click.echo(click.style(msg, fg="red"), err=True)
+                    exit(1)
 
+            # When Error occurs, return the test name as it is passed.
+            subset_result = None
```

Comment:
Unnecessary initialization. In Python, a block statement like `if` does not create a new variable scope.